app/models/admin/tag_filter.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+class Admin::TagFilter
+  KEYS = %i(
+    status
+    name
+    order
+  ).freeze
+
+  attr_reader :params
+
+  def initialize(params)
+    @params = params.to_h.symbolize_keys
+  end
+
+  def results
+    scope = Tag.reorder(nil)
+
+    params.each do |key, value|
+      next if key.to_s == 'page'
+
+      scope.merge!(scope_for(key, value)) if value.present?
+    end
+
+    scope
+  end
+
+  private
+
+  def scope_for(key, value)
+    case key.to_s
```

Comment:
Same as above: Keys should be symbols, why do the extra step of converting them?